### ISSUE TITLE
When you borrow a book and then return it, it should take you back to the page of the book

### ISSUE THREAD

**Issue by RayBB** [internetarchive/openlibrary#9115]
### Describe the problem that you'd like solved

Right now, if you borrow a book and return it you will be redirected to https://openlibrary.org/account/loans, which doesn't even mention the book you have returned. It is a rather confusing experience to me.

### Proposal & Constraints

I think we should return the user to the page of the book they just requested.

At the very very least, it should take you to the loan history page (https://openlibrary.org/account/loan-history) which at least shows your recent loans.

**Comment by mekarpeles** [comment]
Regarding returning the patron to the loans page v. the book page, I think we used to return to the book page and changed it because there may have been assumption if they were done reason one book, they may want to return to the rest of their loans. Agreed that it would be nice for this page to also show some of their recently returned books. I think this is actually a better solution (updating the loan page v. going to the book page), though discussion welcome during community call

**Comment by mekarpeles** [comment]
I think we also now have this capability as we have the loan history page which has an API.

**Comment by bhardwajparth51** [comment]
Hi, @RayBB I dug into the return flow to understand what's happening.

`ReturnForm.html` sets `redirect` to `$request.fullpath`, so when you return a book from `/account/loans`, you end up back there — which is working as coded, just not ideal UX.

From the earlier discussion, it sounds like the preferred fix is improving `/account/loans` itself rather than changing the redirect. I think the cleanest way to do that would be:

- A small tweak in `borrow.py` to append `?returned=<edition_id>` to the redirect URL after a successful return.
- On `/account/loans`, if that param is present, show a dismissible banner like: "You've successfully returned [Book Title]."
- A small "Recently Returned" section pulling from the existing loan history API could give users context even without the query param.

This keeps the redirect behaviour the same, but makes the UX much clearer. Happy to draft an implementation plan and open a PR if this direction works.

**Comment by RayBB** [comment]
Wanna try to create a proof of concept PR for this? @lokesh might have some design opinions here on the dismissable part

**Comment by bhardwajparth51** [comment]
@RayBB Yes, I'd be happy to create a proof of concept PR! I'll start by implementing the backend tweak in `borrow.py` to append the `?returned=<edition_id>` parameter and putting together the "Recently Returned" section on `/account/loans` using the existing loan history API.
Quick questions on scope:

1. For the "Recently Returned" section, should I show the last 3-5 books, or would you prefer a different number?
2. Should this section be visible all the time on `/account/loans`, or only appear if you have loan history?
3. I noticed there's an existing dismissible banner component (`page-banner--dismissable`) in the codebase - should I use that for the "Successfully returned [Book Title]" message for the POC?

Happy to iterate on design once we have the functionality working

**Comment by RayBB** [comment]
Let's skip the recently returned. I don't think we should actually do it because it adds more design with.
And yes try that banner component.

**Comment by bhardwajparth51** [comment]
Got it. I'll skip the recently returned section and use the `page-banner--dismissable` component. Will open a POC PR shortly

**Comment by lokesh** [comment]
A page banner feels heavy for this confirmation message. A flash-message which feels more connected to the page, rather than the site feels more appropriate.

<img width="922" height="168" alt="Image" src="https://github.com/user-attachments/assets/f4378e6c-fcf1-431e-a639-2795c6f59fe8" />

https://github.com/internetarchive/openlibrary/wiki/Design-Pattern-Library#flashmessage
